noobaa/noobaa-core
# delegation removal for s3

### Explain the changes
1. due to performance issues remove call to s3 delegations
2. replaced the call for delegation with one request for s3 params. to be later used for all next read/write operations
3. saving to io_usage in the block_store_client, periodically updating the agent 
4. other performance fixes like bigger watermarks and larger semaphore for s3 agents

### Issues: Fixed #xxx / Gap #xxx
1. Fixed #5731

### Testing Instructions:
1. run the s3cat as seen in issue #5731 - observe better performance than in previous versions
